engine/runtime/src/main/scala/org/enso/compiler/core/IR.scala

```diff
@@ -2074,6 +2074,37 @@ object IR {
         * @return `true` if the value is fractional, `false` otherwise.
         */
       def isFractional: Boolean = value.contains(".")
+
+      /** Checks the values in the literal converts that to approviate JVM value.
+        * @return Double, Long, BigInteger or String (representing compiler error message)
+        */
+      def numericValue: Any = {
+        if (isFractional) {
+          value.toDouble
+        } else if (base.isDefined) {
+          val baseNum =
+            try {
+              Integer.parseInt(base.get)
+            } catch {
+              case _: NumberFormatException =>
+                return s"Invalid number base $base seen during codegen."
```

Comment:
I'd rather we don't use Strings for a method that has `numericValue` in the name. `numericValue` should return `Either[String,Any]` instead and then you can map on `baseNum`

Reply:
Changed the method to `throw new CompilerError`.